src/main/java/com/synopsys/integration/blackduck/nexus3/task/inspector/InspectorMetaDataProcessor.java

```diff
@@ -67,7 +67,29 @@ public InspectorMetaDataProcessor(CommonMetaDataProcessor commonMetaDataProcesso
 
     public void updateRepositoryMetaData(BlackDuckService blackDuckService, String blackDuckServerUrl, ProjectVersionView projectVersionView, Map<String, AssetWrapper> assetWrapperMap,
         TaskStatus status) throws IntegrationException {
+        String projectVersionHref = projectVersionView.getHref().orElse("MISSING HREF");
+        logger.debug("Checking for components in Project Version: '{}'.", projectVersionHref);
         List<ProjectVersionComponentView> versionComponentViews = commonMetaDataProcessor.checkAssetVulnerabilities(blackDuckService, projectVersionView);
+        if (versionComponentViews.isEmpty()) {
+            logger.error("Could not find components in Project Version: '{}'. Check to see if the Code Locations and scans have finished.", projectVersionHref);
+        } else {
+            logger.debug("Found '{}' components in Project Version: '{}'.", versionComponentViews.size(), projectVersionHref);
+            processAssetMapAndBlackDuckComponents(versionComponentViews, blackDuckServerUrl, projectVersionView, assetWrapperMap, status);
+        }
```

Comment:
Adding this logging to confirm and issue a customer is experiencing because it looks like they are seeing an issue where the logs claim the code locations are uploaded and we are done waiting for them to process but it appears there are no components in the BOM and they skip down to the log messages on line 79 and 81 with no assets matching any components.